### PR TITLE
Terminal backspacing (take 2)

### PR DESCRIPTION
**Pull request by heronhaye**
Linux Demo: https://asciinema.org/a/BfvzbZnYK4imPwKR0d4wVOzrg

Now correctly displays prompt for secrets. Can test on local by setting `$ DISPLAY=` before trying to login.

### CODE REVIEW

**Comment by zapu** [comment]
circle failed because of protocol stuff, needs rebase.

I'll pull the code later today and try various things, but it would be best to watch CI or run kbfs tests against it probably.

File: go/minterm/minterm.go [hunk 1]
```diff
@@ -59,42 +58,34 @@ func (m *MinTerm) Size() (int, int) {
...
+	ret, err := terminal.NewTerminal(m.getReadWriter(), prompt).ReadLine()
 	return ret, convertErr(err)
 }
 
-func (m *MinTerm) readSecret() (string, error) {
+func (m *MinTerm) readSecret(prompt string) (string, error) {
 	m.makeRaw()
 	defer m.restore()
-	ret, err := terminal.NewTerminal(m.getReadWriter(), "").ReadPassword("")
+	ret, err := terminal.NewTerminal(m.getReadWriter(), "").ReadPassword(prompt)
```
[the remark this answers is not in the document]
Comment (zapu) [reply go/minterm/minterm.go]: why does this happen though?

[the remark this answers is not in the document]
Comment (zapu) [reply go/minterm/minterm.go]: ah ok, cause:
```
// ReadPassword temporarily changes the prompt and reads a password, without
// echo, from the terminal.
func (t *Terminal) ReadPassword(prompt string) (line string, err error) {
```
so empty string passed to `ReadPassword` would overwrite the prompt passed to `NewTerminal`. And before any of this we weren't relying on terminal prompts but instead we've just wrote it ourselves.

thanks for working on this!

[the remark this answers is not in the document]
Comment (heronhaye) [reply go/minterm/minterm.go]: Thanks for looking at it Michał. I'll figure out how to run kbfs-server tests and merge if they pass.

**Pull request merged**